hamilton/src/moveit_getCartesianPathService.cpp

```diff
@@ -0,0 +1,160 @@
+        ph.getParam(GROUP_PARAM,group_) &&
+        ph.getParam(WORLD_FRAME_PARAM,world_frame_) &&
+        ph.getParam(TCP_FRAME_PARAM,tcp_frame_) &&
+        ph.getParam(TOOL_SPEED_PARAM,tool_speed_) &&
+        ph.getParam(INTERPOLATION_INTERVAL,interp_interval_)
+        )
+    {
+      ROS_INFO_STREAM("Parameters loaded successfully");
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Parameters failed to load");
+      return false;
+    }
+
+    // loading plugin
+    pluginlib::ClassLoader<descartes_core::RobotModel> descartes_robot_model_class_loader(
+        "descartes_core","descartes_core::RobotModel");
+    pluginlib::ClassLoader<descartes_core::PathPlannerBase> planner_class_loader(
+        "descartes_core","descartes_core::PathPlannerBase");
+    try
+    {
+      robot_model_ = descartes_robot_model_class_loader.createInstance(robot_model_plugin_);
+      planner_ = planner_class_loader.createInstance(planner_plugin_);
+    }
+    catch(pluginlib::PluginlibException& ex)
+    {
+      ROS_ERROR_STREAM("Failed to load plugin: "<<ex.what());
+      return false;
+    }
+
+    // initializing plugins and interpolator
+    if(!robot_model_->initialize(ROBOT_DESCRIPTION_NAME,group_,world_frame_,tcp_frame_))
+    {
+      ROS_ERROR_STREAM("Robot model pluging initialization failed");
+      return false;
+    }
+
+    descartes_core::RobotModelConstPtr rm = robot_model_;
+    if(!(planner_->initialize(rm)
+        && planner_->setConfig(planner_config_)))
+    {
+      ROS_ERROR_STREAM("Planner plugin initialization failed");
+      return false;
+    }
+
+    if(!interpolator_.initialize(robot_model_,tool_speed_,interp_interval_,zone_radius_))
+    {
+      ROS_ERROR_STREAM("Interpolator initialization failed");
+      return false;
+    }
+
+
+    cartesian_path_server_ = nh.advertiseService(GET_CARTESIAN_PATH_SERVICE,
+                                                 &CartesianPathService::cartesianPathCallback,
+                                                 this);
+
+
+    return true;
+  }
+
+
+  bool cartesianPathCallback(moveit_msgs::GetCartesianPath::Request& req,
+                             moveit_msgs::GetCartesianPath::Response& res)
+  {
+
+
+    return true;
+  }
+
+protected:
+
+  descartes_core::PathPlannerBasePtr planner_;
+  descartes_core::RobotModelPtr robot_model_;
+  descartes_trajectory::CartesianInterpolator interpolator_;
+
+  ros::ServiceServer cartesian_path_server_;
+
+  std::string robot_model_plugin_;
+  std::string planner_plugin_;
+  std::string group_;
+  std::string world_frame_;
+  std::string tcp_frame_;
+  std::map<std::string,std::string> planner_config_;
+  double tool_speed_;
+  double interp_interval_;
+  double zone_radius_;
+
+};
+
+
+int main(int argc,char** argv)
+{
+  ros::init(argc,argv,"cartesian_path_service");
+  ros::NodeHandle nh;
+
+  CartesianPathService cp;
+  if(!cp.run())
+  {
+    return -1;
+  }
+
+  return 0;
+}
```

Comment:
The name of the file does not follow naming convention.  See [here](https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#File_Names)